### ISSUE TITLE
permissions

### ISSUE THREAD

**Issue by lamboap** [flipboxfactory/saml-sp#130]
Hi,

I was successful in having a pre-created account on CraftCMS use my test Okta account for SSO as an admin.
I'm currently working through the workflow of an Okta account accessing Craft for the first time.  Will I need to have a group set up with the correct permissions on Craft in advance so when the newly created account on Craft is made they'll be assigned to the correct group with the proper permissions.  

I was getting failure messages until I added the user into CraftCMS (first, last, email address) in advance and then access exceptions until I logged in with an admin account and gave the user the proper permissions.

The goal would be to have a newly created account from an Okta login have the correct permissions for an 'editorial' group (as an example).

thanks,
Lambert

**Comment by lamboap** [comment]
Hi,

We've created a new account on Okta and assigned them to the CraftCMS app.  Logging into Okta and then selecting Craft will attempt to create an account with Craft but it throws this exception
```
name: "Exception",
message: "User save failed: {"email":["Email cannot be blank."]}",
```

We've made sure these fields were sent with the assertion.
```ATTRIBUTE STATEMENTS
Name Name Format Value
firstName Unspecified user.firstName
lastName Unspecified user.lastName
email Unspecified user.email```

Only after adding the account on the CraftCMS side did it allow the user to be logged in by not authorized to see anything throwing a 403 forbidden error until they were added to the group in CraftCMS.

thanks,
Lambert

**Comment by dsmrt** [comment]
Look at the following documentation: https://saml-sp.flipboxfactory.com/configure/settings.html#group-configuration-group-assignment

and there is some more info in the this blog post on syncing groups from Okta: https://www.flipboxdigital.com/insights/configuring-saml-sp-with-okta-com

Basically, you will send the attribute "groups" over from Okta and if the group already exists in Craft, the plugin will assign the user to the group.

**Comment by lamboap** [comment]
Thanks @dsmrt I've created the group on 0kta and also created a "group site" under the same site in craft cms because we are using GatsbyJS and CraftCMS in headless mode.  The new user group includes rights to edit the original site and not the new "group site" which is solely used for SSO.

**Comment by lamboap** [comment]
If a user is added to an additional 0kta group after they've been to CraftCMS which also has the same named group in Craft would they be added there as well after a subsequent log-in?  
Similarly, If we remove them from a group in 0kta (they roll off the project in Craft) would an attempt to log into Craft from 0kta remove them from the group there as well?

**Comment by dsmrt** [comment]
Question 1) Yes, if the new group is passed in the SAML Response on login and the group already exists in Craft (and the SAML config is setup correctly), the user will be added to the group in craft.

Question 2) This actually not how it works right now but it probably should (maybe with opt-in with a config). We merge the existing groups of the user with the new groups found from in the SAML Response. I'd argue this might be backward and the default behavior should rely on SAML SSO to manage the groups. 

I added this issue for no. 2: https://github.com/flipboxfactory/saml-sp/issues/133
I can probably add this in the next couple weeks to a month so understand that may take some time.

**Comment by lamboap** [comment]
hi @dsmrt 

Question 1) Yes, if the new group is passed in the SAML Response on login and the group already exists in Craft (and the SAML config is setup correctly), the user will be added to the group in craft.
< this is great.  Our use-case would be:  a user account that was created as a writer/author and then possibly promoted to be an editor or even added to the admin groups in 0kta.  That transition would be seamless on the Craft side.     

"the default behavior should rely on SAML SSO to manage the groups."
that would be ideal for our use-case as the admin for 0kta, ideally doesn't have access to Craft or wouldn't want to know how to administer users in Craft CMS.

thanks,
Lambert

**Comment by dsmrt** [comment]
Note that although it's not terribly efficient, you can edit the users group association with this event: https://saml-sp.flipboxfactory.com/configure/events.html#assign-user-to-a-user-group-based-on-a-property

Although, it'd be better to have an event that is easier to work with and/or be able to toggle a config to opt-in to the merging of the groups.

I just wanted to point out that if you need to, you can fix this with the event above.

**Comment by dsmrt** [comment]
If you update to 2.7.1 you can see some changes including the `mergeExistingGroups` config. More info can be found here:
https://saml-sp.flipboxfactory.com/configure/groups.html#user-group-saml-sp-configs

**Comment by ilicmarko** [comment]
Not sure if I should open a new issue or use this one.

How do we assign the user to `Admin`? @dsmrt 

I receive the `groups` attribute and it has `admin` but that is not [Craft's special admin user](https://craftcms.com/docs/3.x/user-management.html#admin-accounts). I have to create a new user group with handle `admin` and that will work. The downside to this is I always have to enable content changes for each new entry/section...

Is there some special value that SAML can return inside to group to use the Craft admin (example `craft_admin`)?

**Comment by dsmrt** [comment]
The plugin doesn’t support setting the user with admin privileges automatically. This is logic you’ll have to add to a group, then add the user to the group or use the event as mentioned here: https://saml-sp.flipboxfactory.com/configure/settings.html#group-configuration-group-assignment
